### PR TITLE
mgmt, fix permission.dataActions

### PR DESCRIPTION
**Pull request by XiaofeiCao**
# Description

issue: https://github.com/Azure/azure-sdk-for-java/issues/33365

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**
## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-samples/src/main/java/com/azure/resourcemanager/samples/Utils.java [hunk 1]
```diff
@@ -2581,6 +2581,10 @@ public static void print(RoleDefinition role) {
             for (String notAction : permission.notActions()) {
                 builder.append("\n\t\t\tName :").append(notAction);
             }
+            builder.append("\n\t\tPermission Data Actions: " + permission.dataActions().size());
+            for (String notAction : permission.dataActions()) {
```
Comment (haolingdong-msft) [inline sdk/resourcemanager/azure-resourcemanager-samples/src/main/java/com/azure/resourcemanager/samples/Utils.java:2585]: `for (String notAction : permission.dataActions())` -> `for (String dataAction : permission.dataActions())`

File: sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/implementation/PermissionImpl.java [hunk 2]
```diff
@@ -26,7 +26,7 @@ public List<String> notActions() {
 
     @Override
     public List<String> dataActions() {
-        return innerModel().actions();
+        return innerModel().dataActions();
     }
 
     @Override
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/implementation/PermissionImpl.java:32]: Is "notDataActions" below also not correct?

File: sdk/resourcemanager/azure-resourcemanager-samples/src/main/java/com/azure/resourcemanager/samples/Utils.java (hunk 1 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-samples/src/main/java/com/azure/resourcemanager/samples/Utils.java]: Changed, thanks!

**Review by weidongxu-microsoft — APPROVED** [review]

File: sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/implementation/PermissionImpl.java (hunk 2 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/implementation/PermissionImpl.java:32]: They're not correct. Fixed as well. Thanks!

**Pull request merged**